pkg/keycloak/phaseHandler.go

```diff
@@ -95,6 +96,108 @@ func (ph *phaseHandler) Accepted(sso *v1alpha1.Keycloak) (*v1alpha1.Keycloak, er
 	return kc, nil
 }
 
+// Upgrade will perform the necessary actions to do the upgrade for an in place sso instance
+// Note we rely on imagestreams that the operator itself does not install (these are installed currently by the installer in the openshift ns )
+func (ph *phaseHandler) Upgrade(sso *v1alpha1.Keycloak) (*v1alpha1.Keycloak, error) {
+	cpSSO := sso.DeepCopy()
+	dc, err := ph.ocDCClient.DeploymentConfigs(cpSSO.Namespace).Get(SSO_APPLICATION_NAME, v12.GetOptions{})
+	if err != nil {
+		return cpSSO, errors.Wrap(err, "failed to get current deploymentconfig for sso")
+	}
+	if !CanUpgrade(cpSSO.Status.Version) || (cpSSO.Status.Phase != v1alpha1.PhaseUpgrading && cpSSO.Status.Phase != v1alpha1.PhaseReconcile) {
+		logrus.Debug("not upgrading from version ", cpSSO.Status.Version, " in phase ", cpSSO.Status.Phase)
+		return nil, nil
```

Comment:
I'd be inclined to set the state in the CR here to something like:
"willNotUpgrade" and a reason like "no upgrade path from X to 7.3.2"